workers/price-reporter/src/exchange/binance.rs

```diff
@@ -65,15 +65,34 @@ pub struct BinanceConnection {
 
 impl BinanceConnection {
     /// Construct the websocket url for the given asset pair
-    fn websocket_url(base_token: &Token, quote_token: &Token) -> Url {
-        let base_ticker = base_token.get_exchange_ticker(Exchange::Binance);
-        let quote_ticker = quote_token.get_exchange_ticker(Exchange::Binance);
-        Url::parse(&format!(
+    fn websocket_url(
+        base_token: &Token,
+        quote_token: &Token,
+    ) -> Result<Url, ExchangeConnectionError> {
+        let base_ticker = base_token.get_exchange_ticker(Exchange::Binance).ok_or(
+            ExchangeConnectionError::UnsupportedPair(
+                base_token.clone(),
+                quote_token.clone(),
+                Exchange::Binance,
+            ),
+        )?;
```

Comment:
we do this exact logic in a few places, probably best to just write a helper for it